### ISSUE TITLE
noUnusedLocals not working with array destructuring assignment in for-loop

### ISSUE THREAD

**Issue by AnyhowStep** [microsoft/TypeScript#33054]
**TypeScript Version:**  3.5.1

**Search Terms:**

+ noUnusedLocals
+ array destructuring assignment
+ for-loop

**Code**

```ts
//--noUnusedLocals enabled

//Expected: No Error
//Actual  : '_a' is declared but its value is never read.
for (const [_a, b] of []) {
    console.log(b);
}

//Expected: No Error
//Actual  : No Error
function foo ([_a, b] : [any, any]) {
    console.log(b);
}

//Expected: No Error
//Actual  : No Error
const [_a, b] = [1, 1];
console.log(b);
```

**Expected behavior:**

The `_a` should not have an error in,
```ts
for (const [_a, b] of []) {
/*snip*/
```

**Actual behavior:**

It errors

**Playground Link:** 

[Playground](http://www.typescriptlang.org/play/?noUnusedLocals=true#code/PTC0oOwewVQhXAzgUwCYBkoGMCGAbRAAmQhwCM80BYAKFpAFEAPAB2SwBc0AuQgOSiEGAJ2FRh9YAEFO8fIUK8A5AH0cSwgEsiqdnhzC0hMvA5aORAG754yLUQjJLyYYUM5UAOloAzcYQAKLCgIRDMAbTUAGmMAXUIoH0Jw2IBKQgBvWgUFYNCoSk88KABzALJUgG5aAF9aSWY2Th5+QRExCRoQGQ45PAVeASFRcV94CE5NEMI-QQDInBiyeN5wnAgATxj1jbTM7Jy8xALkItLyqtr6ruBG9i5UQbaRzu7ZeUVW4Y7aI4jouKEAC8yQAjDFQbFqjQjiczmUKpUgA)

**Related Issues:** 

### TODO

**Comment by RyanCavanaugh** [comment]
This is kind of arguable since you could have just omitted the element entirely, but we should be consistent and also not have an arbitrary breaking change

**Comment by AnyhowStep** [comment]
It just makes code a little more readable for me =x

But I guess this might be better,
```ts
for (const [/*a*/, b] of []) {
/*snip*/
```

So I get an identifier for what was omitted and I don't accidentally use `a` or `_a`.
(I think I've seen this used in the TS code base a bunch of times in function calls)
